message-relay/service-worker.js

```diff
@@ -1,23 +1,26 @@
 // Listen for messages from clients.
 self.addEventListener('message', function(event) {
-  // Get all the connected clients and forward the message along.
-  self.clients.matchAll().then(function(clientList) {
-    clientList.forEach(function(client) {
-      // Skip sending the message to the client that sent it.
-      if (client.id === event.source.id) {
-        return;
-      }
-      client.postMessage({
-        client: event.source.id,
-        message: event.data
+  event.waitUntil(
```

Comment:
Each `letter` triggers a `TypeError: event.waitUntil is not a function` error here.